DirectProgramming/C++/CompilerInfrastructure/OpenMP_Offload_Features/src/usm_and_composability_with_dpcpp.cpp

```diff
@@ -0,0 +1,67 @@
+//==============================================================
+// Copyright © 2021 Intel Corporation
+//
+// SPDX-License-Identifier: MIT
+// =============================================================
+
+#include <omp.h>
+#include <CL/sycl.hpp>
+#include <algorithm>
+#include <iostream>
+
+extern "C" void *omp_target_get_context(int);
+
+#ifdef OCL_BACKEND
+#pragma omp requires unified_shared_memory
+#endif
+
+int main() {
+  const unsigned Size = 200;
+  int D = omp_get_default_device();
+#ifdef OCL_BACKEND
+  cl::sycl::queue Q(
+      cl::sycl::context(static_cast<cl_context>(omp_target_get_context(D))),
+      cl::sycl::gpu_selector());
+#else
+  cl::sycl::queue Q;
+#endif
+  std::cout << "SYCL: Running on "
+            << Q.get_device().get_info<cl::sycl::info::device::name>() << "\n";
+  if (!Q.get_device()
+           .get_info<cl::sycl::info::device::usm_shared_allocations>()) {
+    std::cout << "SYCL: USM is not available\n";
+    return 0;
+  }
+  auto validate = [](int *Data) {
+    for (unsigned I = 0; I < Size; ++I)
+      if (Data[I] != 100 + I) return "failed";

```

Comment:
The return from main should be 0 (succes) or non-zero (failure).   You should not be passing character array pointers.